packages/ifpack2/src/Ifpack2_PowerMethod.hpp

```diff
@@ -152,70 +155,142 @@ powerMethodWithInitGuess (const OperatorType& A,
      "compute the initial guess), or because the norm2 method has a "
      "bug.  The first is not impossible, but unlikely.");
 
-  if (verbose) {
-    *out << "  Original norm1(x): " << x->norm1 ()
+  if(verbose) {
+    *out << "  Original norm1(x): " << x->norm1()
           << ", norm2(x): " << norm << endl;
   }
 
-  x->scale (one / norm);
+  x->scale(one / norm);
 
-  if (verbose) {
-    *out << "  norm1(x.scale(one/norm)): " << x->norm1 () << endl;
+  if(verbose) {
+    *out << "  norm1(x.scale(one/norm)): " << x->norm1() << endl;
   }
 
   if(y.is_null())
-    y = Teuchos::rcp(new V(A.getRangeMap ()));
-
-  for (int iter = 0; iter < numIters-1; ++iter) {
-    if (verbose) {
-      *out << "  Iteration " << iter << endl;
+    y = Teuchos::rcp(new V(A.getRangeMap()));
+
+  // GH 04 Nov 2022: The previous implementation of the power method was incorrect.
+  // It computed x->norm2() inside the loop, but returned x^T*Ax. This returned horribly
+  // incorrect answers in certain cases, such as the Cheby_belos test, which has two 
+  // largest eigenvalues of 12.5839 and -10.6639. The power method iteration history 
+  // would appear to converge to something between 10 and 12, but the result would 
+  // return something between -10 and 12 instead. These have now been split into two
+  // routes which will behave consistently to themselves.
```

Comment:
I'm not sure this comment makes much sense without the actual matrix.

Reply:
It references the test `Cheby_belos`, which has the matrix it refers to, right?

Reply:
Yep.